src/audio/kpb.c

```diff
@@ -772,9 +816,17 @@ static int kpb_prepare(struct comp_dev *dev)
 		kpb->clients[i].r_ptr = NULL;
 	}
 
+#if CONFIG_AMS
+	/* AMS Register KPB for notification */
+	ret = ams_helper_register_consumer(dev, &kpb->kpd_uuid_id,
+					   (const uint8_t *)ams_kpd_msg_uuid,
```

Comment:
is the cast necessary? It's a classic with UUIDs to keep cast pointers, can we not use a 'native' uuid_t or something that everyone can agree on?

Reply:
See below.